src/lib/spinel/radio_spinel.cpp

```diff
@@ -1805,6 +1810,23 @@ otError RadioSpinel::Transmit(otRadioFrame &aFrame)
 
     mTransmitFrame = &aFrame;
 
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT && OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    if (mTransmitFrame->mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0)
+    {
+        uint8_t *timeIe         = mTransmitFrame->mPsdu + mTransmitFrame->mInfo.mTxInfo.mIeInfo->mTimeIeOffset;
+        uint64_t offsetPlatTime = static_cast<uint64_t>(static_cast<int64_t>(otPlatTimeGet()) +
+                                                        mTransmitFrame->mInfo.mTxInfo.mIeInfo->mNetworkTimeOffset);
```

Comment:
The `otPlatTimeGet()` returns the host side time (not the value on RCP).

May be worth checking whether time used here need to be the RCP time or host time?

Reply:
It would be more precise to use the RCP time (if available) and to use the expected time to send (CSL). Will work on a solution. Thanks.

Reply:
I now get the radio time (if available), and if available, I schedule a transmit with the radio time + an offset, the same offset time I transmit in the sync message.